stdlib/array.gr

```diff
@@ -506,29 +518,31 @@ provide let fill = (value, array) => {
  * @param stop: The (exclusive) index to end replacement
  * @param array: The array to update
  *
- * @throws Failure(String): When the start index is out of bounds
- * @throws Failure(String): When the start index is greater then the stop index
+ * @throws IndexOutOfBounds: When the start index is out of bounds
+ * @throws IndexOutOfBounds: When the start index is greater then the stop index
  *
  * @since v0.2.0
  */
 provide let fillRange = (value, start, stop, array) => {
   let length = length(array)
-  let startIndex = if (start < 0) length + start else start
-  let stopIndex = if (stop < 0) length + stop else stop
+  let startIndex = wrapNegativeIndex(length, start)
+  let stopIndex = wrapNegativeIndex(length, stop)
+  // Ensure we aren't working with an `stop` value that is too big
+  let stopIndex = if (stopIndex < 0 || stopIndex > length) {
+    length
+  } else {
+    stopIndex
+  }
```

Comment:
Just making sure we want this specific behavior. We don't want this to do "cycles" like Array.rotate, right? It's more like Array.slice?

Reply:
Talked about this on Community call and this looks correct. Will resolve this and approve later.